### PR TITLE
Add a couple of static tests

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9335893d323d0832dec984b8d10d3a45df3082fe-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9335893d323d0832dec984b8d10d3a45df3082fe-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9335893d323d0832dec984b8d10d3a45df3082fe-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9335893d323d0832dec984b8d10d3a45df3082fe-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-4c0258b76be74aa0aae73a11271f60b9c94bce41-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-4c0258b76be74aa0aae73a11271f60b9c94bce41-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-4c0258b76be74aa0aae73a11271f60b9c94bce41-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-4c0258b76be74aa0aae73a11271f60b9c94bce41-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-d28b09ab306ea4e0436d8edcd00df6c40c456372-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-d28b09ab306ea4e0436d8edcd00df6c40c456372-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-d28b09ab306ea4e0436d8edcd00df6c40c456372-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-d28b09ab306ea4e0436d8edcd00df6c40c456372-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-fb41acb6cf83f0244049090490044a7db69ca037-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-fb41acb6cf83f0244049090490044a7db69ca037-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-fb41acb6cf83f0244049090490044a7db69ca037-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-fb41acb6cf83f0244049090490044a7db69ca037-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9acd12cf8201fc629b9b0f5591690ab639011b29-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9acd12cf8201fc629b9b0f5591690ab639011b29-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9acd12cf8201fc629b9b0f5591690ab639011b29-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-9acd12cf8201fc629b9b0f5591690ab639011b29-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-a28ea138652ded2912136b70c0d84dac714aa967-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-a28ea138652ded2912136b70c0d84dac714aa967-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-a28ea138652ded2912136b70c0d84dac714aa967-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-a28ea138652ded2912136b70c0d84dac714aa967-x86_64.apk)

**Comment by m-kuhn** [comment]
@3nids can I get your approval here?

File: scripts/prepare-commit.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#!/bin/bash
+#!/usr/bin/env bash
 ###########################################################################
-#    prepare-commit.sh
+#    prepare_commit.sh
```
Comment (3nids) [inline scripts/prepare-commit.sh:3]:
```suggestion
#    prepare-commit.sh
```

File: scripts/prepare-commit.sh [hunk 2]
```diff
@@ -14,8 +14,21 @@
...
+TOPLEVEL=$(git rev-parse --show-toplevel)
 
-PATH=$(git rev-parse --show-toplevel)/scripts:$PATH
+PATH=$TOPLEVEL/scripts:$PATH:$PWD/scripts
+
+if ! tty -s && [[ "$0" =~ /pre-commit ]]; then
+    exec </dev/tty
+fi
+
+cd "$TOPLEVEL"
```
Comment (3nids) [inline scripts/prepare-commit.sh:25]: no push/popd ?

File: scripts/spell_check/spell_test.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  FILES=$(git diff --diff-filter=AM --name-only ${TRAVIS_COMMIT_RANGE/.../..} | tr '\n' ' ' )
+  export PATH=${HOME}/osgeo4travis/bin:${PATH}
+else
+  echo "Running spell test on all files (might take a while)"
+  FILES=""
+fi
+
+if [[ -n $FILES ]]; then
+  DIR=$(git rev-parse --show-toplevel)/scripts/spell_check
+  ${DIR}/check_spelling.sh -r $FILES
```
Comment (3nids) [inline scripts/spell_check/spell_test.sh:24]:
```suggestion
  scripts/spell_check/check_spelling.sh -r $FILES
```

File: scripts/prepare-commit.sh (hunk 2 above)
Comment (m-kuhn) [reply scripts/prepare-commit.sh:25]: It's a copy from qgis, I guess it should be good

File: scripts/spell_check/spell_test.sh (hunk 3 above)
Comment (m-kuhn) [reply scripts/spell_check/spell_test.sh:24]: `./scripts/...` (note the leading dot)?

Comment (3nids) [reply scripts/spell_check/spell_test.sh:24]: as you prefer I guess

File: src/core/featuremodel.h [hunk 4]
```diff
@@ -125,7 +125,11 @@ class FeatureModel : public QAbstractListModel
      * Will reset the feature to the original values and dismiss any buffered edits.
      */
     Q_INVOKABLE void reset();
-    Q_INVOKABLE bool create();
+
+    /**
+     * Will create this feature as a new feature on the data source
+     */
+    Q_INVOKABLE void create();
```
Comment (m-kuhn) [inline src/core/featuremodel.h:132]:
```suggestion
    Q_INVOKABLE bool create();
```

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-0e67b8970f90d5ddb18165a6a049ffa74e4e19bf-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-0e67b8970f90d5ddb18165a6a049ffa74e4e19bf-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-0e67b8970f90d5ddb18165a6a049ffa74e4e19bf-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1002-0e67b8970f90d5ddb18165a6a049ffa74e4e19bf-x86_64.apk)

**Pull request merged**